### PR TITLE
Unify bookmarks context menu across different views

### PR DESCRIPTION
**Pull request by mallexxx**
Task/Issue URL: https://app.asana.com/0/1202406491309510/1207775978136617/f
Project: https://app.asana.com/0/1202406491309510/1207775978136617/f
Includes: 
https://github.com/duckduckgo/macos-browser/pull/3106
https://github.com/duckduckgo/macos-browser/pull/3107
https://github.com/duckduckgo/macos-browser/pull/3110
https://github.com/duckduckgo/macos-browser/pull/3111
https://github.com/duckduckgo/macos-browser/pull/3138
https://github.com/duckduckgo/macos-browser/pull/3112
https://github.com/duckduckgo/macos-browser/pull/3113

**Description**:
- a follow-up for https://github.com/duckduckgo/macos-browser/pull/3106 discussion to unify bookmarks context menus to reduce code duplication

**Steps to test this PR**:
1. Validate all menu items work the same way as before in the Bookmarks bar (for a bookmark, folder, empty space)
2. Bookmark Manager - tree view (for a folder and empty space)
3. Bookmark Manager - main view (for bookmarks, folders, nested folders, empty space, from "…" button)
4. Bookmarks Popover - (for bookmarks, folders, nested folders, empty space, from "…" button)

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -75,7 +75,7 @@
     {
       "identity" : "lottie-spm",
       "kind" : "remoteSourceControl",
-      "location" : "https://github.com/airbnb/lottie-spm.git",
+      "location" : "https://github.com/airbnb/lottie-spm",
```
Comment (jotaemepereira) [inline DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:78]: 🔍 This should be reverted

**Review by jotaemepereira — CHANGES_REQUESTED** [review]
@mallexxx This is looking good. I’m requesting changes because I found the following:

- Tapping the three dots on a bookmark on the bookmarks manager does not open the menu
- The `Manage Bookmarks` menu item should be disabled when the menu is opened from a bookmark on the bookmarks manager (at the moment, it is enabled)
- Adding a folder is not working as expected. If I have Folder > Bookmark, and I select the folder in the Bookmarks Manager and then tap the three dots to create a folder, the folder is created in the root while it should be created inside the Folder (this works correctly in the panel but not in the manager)
- If you have Folder > Bookmark and select the Bookmark menu and tap on `Move to end` it will move it to the end but in the root folder, instead of moving it inside the current folder.

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (mallexxx) [reply DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: I guess it shouldn‘t as it‘s without `.git` in `Package.swift`

**Review by ayoy — APPROVED** [review]
I've retested issues found by @jotaemepereira and reviewed the code added after his review. All looks good to me and everything works as expected now. Thanks a lot @mallexxx!

**Pull request merged**